### ISSUE TITLE
arm64 LDADD instruction treated as writing output register before reading input register

### ISSUE THREAD

**Issue by comex** [Vector35/binaryninja-api#5603]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5422-dev (19064227)
- OS: macOS
- OS Version: 14.5
- CPU Architecture: M1

**Bug Description:**
The arm64 instruction `LDADD Xs, Xt, [Xn]` atomically adds `Xt` to the memory at `[Xn]` and then returns the old value in `Xs`.  If `Xs` and `Xt` are the same register, then it uses the previous value of the register as the amount to add.  In other words, it first reads from the register `Xt` and then writes to the register `Xs`.  Binary Ninja models this in the other order, leading to incorrect results.

**Steps To Reproduce:**
Load [this object](https://github.com/user-attachments/files/15813501/test.zip).
Disassembly:
```
00000000  200020f8   ldadd   x0, x0, [x1]
00000004  c0035fd6   ret     
```
LLIL:
```
   0 @ 00000000  x0 = [x1].q
   1 @ 00000000  [x1].q = x0 + x0
   2 @ 00000004  <return> jump(x30)
```
Note that x0 is written before being read.  HLIL then turns this into:
```
00000000      int64_t result = *arg2
00000000      *arg2 = result * 2
00000004      return result
```

**Expected Behavior:**
A correct decompilation would be something like:
```
00000000      int64_t result = *arg2
00000000      *arg2 = result + arg1
00000004      return result
```

**Comment by galenbwill** [comment]
While we cannot currently model or represent the atomicity (never mind the acquire/release semantics) of these instructions, it's easy to fix the issue you describe using a temp register. I will add it to the queue of instructions I'm currently in the process of fixing.

**Comment by galenbwill** [comment]
Should be fixed in builds >= 4.1.5500-dev

Note that I fixed all the `LDADD*` variants and added lifting for the `STADD*` aliases. (And as I mentioned, this does not deal with, annotate, or otherwise represent the acquire/release memory ordering semantics.)

The given test.o:

LLIL:
```C++
   0 @ 00000000  temp0 = zx.q([x1].q)
   1 @ 00000000  [x1].q = x0 + zx.q(temp0.q)
   2 @ 00000000  x0 = zx.q(temp0.q)
   3 @ 00000004  <return> jump(x30)
```
MLIL:
```C++
   0 @ 00000000  result = zx.q([arg2].q)
   1 @ 00000000  [arg2].q = arg1 + zx.q(result)
   2 @ 00000000  x0 = zx.q(result)
   3 @ 00000004  return x0
```
HLIL:
```c++
00000000      uint64_t result = *arg2
00000000      *arg2 = arg1 + result
00000004      return result
```
